invenio_rdm_records/services/services.py

```diff
@@ -376,25 +407,36 @@ def delete_pid(self, id_, identity, pid_type, pid_client=None):
         # Permissions
         self.require_permission(identity, "manage", record=draft)
 
-        provider = self.get_provider(pid_type, client_name=pid_client)
+        providers = self.config.pids_providers[pid_type]
+        provider_name, provider_config = self._get_managed_provider(providers)
+
+        is_enabled = provider_config.get("enabled", True)
+        if not is_enabled:
+            raise Exception(f"The provider {provider_name} is not enabled")
+
+        provider = self.get_provider(pid_type, provider_name=provider_name,
+                                     client_name=pid_client)
         pid_attr = draft.pids.get(pid_type)
 
         if not pid_attr:
             raise ValidationError(
-                message=_("Identifier value or client not given for PID " +
-                          f"type {pid_type}"),
+                message=_(f"No registered PID found for type {pid_type}"),
                 field_name="pids",
```

Comment:
minor: `pids.pid_type? or not because it might not be and therefore generic `pids` field

Reply:
I would keep this check before getting the `pid` to make sure everything is correct.
The pid `.get` will fail anyway if not found.

Reply:
You are right, fixing it.